hibernate-reactive-core/src/main/java/org/hibernate/reactive/impl/TransactionalSqlClientConnection.java

```diff
@@ -0,0 +1,71 @@
+package org.hibernate.reactive.impl;
+
+import io.vertx.sqlclient.Pool;
+import io.vertx.sqlclient.SqlConnection;
+import org.hibernate.reactive.util.impl.CompletionStages;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A transaction-capable reactive connection based on Vert.x's
+ * {@link SqlConnection}.
+ */
+public class TransactionalSqlClientConnection extends SqlClientConnection {
+
+	private SqlConnection connection;
+//	private Transaction transaction;
+
+	public TransactionalSqlClientConnection(Pool pool, boolean showSQL) {
+		super(pool, showSQL);
+	}
+
+	protected CompletionStage<SqlConnection> connection() {
+		if ( connection != null ) {
+			return CompletionStages.completedFuture( connection );
+		}
+		else {
+			return toCompletionStage(
+					handler -> pool.getConnection(
+							ar -> {
+								if ( ar.succeeded() ) {
+									connection = ar.result();
+								}
+								handler.handle( toFuture(ar) );
+							}
+					)
+			);
+		}
+	}
+
+	@Override
+	public CompletionStage<Void> beginTransaction() {
+//		return connection().thenApply( conn -> {
+//			transaction = conn.begin();
+//			return null;
+//		});
+		return execute("begin");
```

Comment:
we'll need a way of no-opping this for DB2 because DB2 implicitly starts transactions and does not support the `BEGIN` keyword (commit and rollback are fine though)

Reply:
Oh, ok. Is it better if I use the commented-out version?

Reply:
yea, that should work

Reply:
Alright, thanks, I've made that change.